File: sdks/sdkwork-control-plane-sdk/sdkwork-control-plane-sdk-typescript/composed/src/protocol-module.ts
import type { JsonObject } from './types.js';
import type { ControlPlaneSdkContext } from './sdk-context.js';

export class ControlPlaneProtocolModule {
  constructor(private readonly context: ControlPlaneSdkContext) {}

  getGovernance(): Promise<JsonObject> {
    return this.context.backendClient.protocol.getProtocolGovernance();
  }

  getRegistry(): Promise<JsonObject> {
    return this.context.backendClient.protocol.getProtocolRegistry();
  }
}

File: sdks/sdkwork-control-plane-sdk/sdkwork-control-plane-sdk-flutter/composed/lib/src/protocol_module.dart
import 'package:control_plane_backend_sdk/control_plane_backend_sdk.dart';

import 'context.dart';

class ControlPlaneProtocolModule {
  final ControlPlaneSdkContext context;

  ControlPlaneProtocolModule(this.context);

  Future<JsonObject> getGovernance() {
    return context.backendClient.protocol.getProtocolGovernance();
  }

  Future<JsonObject> getRegistry() {
    return context.backendClient.protocol.getProtocolRegistry();
  }
}
